tests/test_backends.py

```diff
@@ -27,12 +27,16 @@ def test_array(self):
         np_mat = _np.array([_np.ones(3), _np.ones(3)])
         self.assertAllCloseToNp(gs_mat, np_mat)
 
-        gs_mat = gs.array([[gs.ones(3)], [gs.ones(3)]])
-        np_mat = _np.array([[_np.ones(3)], [_np.ones(3)]])
+        gs_mat = gs.array([gs.ones(3), gs.ones(3)], dtype=gs.float64)
+        np_mat = _np.array([_np.ones(3), _np.ones(3)], dtype=_np.float64)
```

Comment:
Nice! Although I'm not sure that assertAllCloseToNp also checks that the results are of the same type. Could you maybe add a separate assert that checks the output's types?